pkg/skaffold/deploy/kubectl.go

```diff
@@ -252,6 +219,72 @@ func (k *KubectlDeployer) readRemoteManifest(ctx context.Context, name string) (
 	return manifest.Bytes(), nil
 }
 
-func (k *KubectlDeployer) Render(context.Context, io.Writer, []build.Artifact, string) error {
-	return errors.New("not yet implemented")
+func (k *KubectlDeployer) Render(ctx context.Context, out io.Writer, builds []build.Artifact, filepath string) error {
+	manifests, err := k.renderManifests(ctx, out, builds)
+
+	if err != nil {
+		return err
+	}
+
+	manifestOut := out
+	if filepath != "" {
+		f, err := os.Open(filepath)
```

Comment:
do we need to `close` this in defer?

Reply:
I didn't write this code, just moved it, so I'm not sure. I'll leave it as it for now.